subsys/net/lib/lwm2m_client_utils/lwm2m/lwm2m_firmware.c

```diff
@@ -740,28 +745,40 @@ static int firmware_block_received_cb(uint16_t obj_inst_id, uint16_t res_id, uin
 
 		/* Store Started DFU type */
 		target_image_type_store(obj_inst_id, image_type);
-		ret = dfu_target_init(image_type, 0, total_size, dfu_target_cb);
-		if (ret < 0) {
-			LOG_ERR("Failed to init DFU target, err: %d", ret);
-			goto cleanup;
-		}
+		do {
+			ret = dfu_target_init(image_type, 0, total_size, dfu_target_cb);
```

Comment:
Only if we end up calling `dfu_target_reset()`. It will erase the DFU area and release all the resources.

So if we erase, then init is needed again. This is similar of what `fota_download.c` does.